main.go

```diff
@@ -35,18 +35,27 @@ func main() {
 
 	// init logger
 	logger := log.NewJSONLogger(os.Stdout)
-	logger = log.With(
-		logger,
+	logger = log.With(logger,
 		"ts", log.DefaultTimestampUTC,
+		"caller", log.DefaultCaller,
 	)
-	logger = leveled.NewFromEnv(logger)
-	logger = log.With(logger, "caller", log.DefaultCaller)
+	// set the allowed log level filter
+	switch strings.ToLower(os.Getenv("LOGLEVEL")) {
+	case "debug":
+		level.NewFilter(logger, level.AllowDebug())
+	case "warn":
+		level.NewFilter(logger, level.AllowWarn())
+	case "error":
+		level.NewFilter(logger, level.AllowError())
+	default:
+		level.NewFilter(logger, level.AllowInfo())
```

Comment:
and here.

Also I'd like to change the default level to `level.AllowAll()` and introduce another case for `info`.